YuJeongMin/YuJeongMin211018/CameraControl.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using UniRx;
+using UniRx.Triggers;
+using UnityEngine;
+
+namespace InGame
+{
+    public class CameraControl : MonoBehaviour
+    {
+        private Camera _camera;
+
+        private const float ScreenRatio = 9f / 16f;
+        private const float DefaultSize = 5f;
+        private const float DefaultPosX = -10;
+        private const float Half = 0.5f;
+
+        [SerializeField] private Transform myCharacterTransform;
+        [SerializeField] private Transform anotherCharacterTransform;
+
+        public void SetTarget(Transform my, Transform another)
+        {
+            myCharacterTransform = my;
+            anotherCharacterTransform = another;
+        }
+
+        public void SetTarget(GameObject my, GameObject another)
+        {
+            myCharacterTransform = my.transform;
+            anotherCharacterTransform = another.transform;
+        }
+
+        private void Start()
+        {
+            _camera = GetComponent<Camera>();
+
+            this.UpdateAsObservable().Subscribe(_ =>
+            {
+                var my = myCharacterTransform.position;
+                var another = anotherCharacterTransform.position;
+                
+                transform.position = CameraPosition(my, another);
+                _camera.orthographicSize = CameraSize(my, another);
+            });
+        }
+
+        // 카메라 위치 : 두 캐릭터 위치의 중점
+        private static Vector3 CameraPosition(Vector3 my, Vector3 another)
+        {
+            var xMid = Mathf.Lerp(my.x, another.x, Half);
+            var yMid = Mathf.Lerp(my.y, another.y, Half);
+
+            return new Vector3(xMid, yMid, DefaultPosX);
```

Comment:
```suggestion
            return new Vector3(xMid, yMid, DefaultPosZ);
```
`DefaultPosX` 상수는 z값 위치에 들어가는데 이름은 X인 이유가 있나요? 여러곳에 사용할 수 있는 값이더라도 혼동 될 수 있는 경우 같은 값이더라도 부분부분 필요한 값이 달라질 수도 있기 때문에 상수 이름은 다르게 정의해주는 것이 좋습니다.

Reply:
지적 감사합니다.